tests/versioned/segments.tap.js

```diff
@@ -6,11 +6,29 @@
 'use strict'
 
 const tap = require('tap')
+const semver = require('semver')
 const helpers = require('./helpers')
 const utils = require('@newrelic/test-utilities')
 const TRANSACTION_PREFX = 'WebTransaction/WebFrameworkUri/Nextjs/GET/'
 const SEGMENT_PREFIX = 'Nodejs/Nextjs/getServerSideProps/'
 const MW_PREFIX = 'Nodejs/Middleware/Nextjs/'
+const nextPkg = require('next/package.json')
+
+function getChildSegments(uri) {
+  const segments = [
+    {
+      name: `${SEGMENT_PREFIX}${uri}`
+    }
+  ]
+
+  if (semver.gte(nextPkg.version, '12.2.0')) {
```

Comment:
I'm not sure I understand this. We'll still be seeing calls to middleware in pre 12.2 versions of Next.js, but we won't be seeing what's inside of those calls anymore?

Reply:
right. we don't instrument middleware pre 12.2.0 any longer because it changed significantly and wasn't stable before this